appwrite/appwrite
# Implementing the OAuth2 provider for 42 School in Appwrite

## What does this PR do?

This PR implements the OAuth2 provider for 42 School in Appwrite, allowing users to authenticate with their 42 School credentials and access Appwrite's features seamlessly.

## Test Plan

1. Set up a development environment with Appwrite.
2. Configure the OAuth2 provider for 42 School in the Appwrite dashboard.
3. Attempt to authenticate with 42 School credentials through the Appwrite authentication flow.
4. Verify successful authentication and access to Appwrite's features.

## Related PRs and Issues

- [Related issue](https://github.com/appwrite/appwrite/issues/7773)
- [PR for @appwrite/console](https://github.com/appwrite/console/pull/946)
- [Issue for @appwrite/console](https://github.com/appwrite/console/issues/945)

## Checklist


## Demonstration

https://github.com/appwrite/appwrite/assets/63407055/6d3090b3-df7f-4359-8333-3c194111ad4d
